src/parsing/Parse_metavariable_type.mli

```diff
@@ -0,0 +1,11 @@
+(* Wrap and unwrap type expressions for metavariable type rule.
+ *
+ * Language parsers typically do not support multiple top-level entry points,
+ * and there can be ambiguity when a single identifier is interpreted as both
+ * an expression and a type. To avoid unexpected behavior caused by parsing
+ * conflicts, we wrap the input string into an expression to explicitly extract
+ * the type and later unwrap it after parsing.
```

Comment:
maybe you can give a concrete example, like for Java we use (%s x) to reuse the cast syntax where x does not matter.